lib/dio/forwarders/string_hash_forwarder.rb

```diff
@@ -0,0 +1,60 @@
+require 'delegate'
+
+module Dio
+  module Forwarders
+    # Pattern Matching relies on Symbol keys for Hash matching, but a significant
+    # number of Ruby Hashes are keyed with Strings. This forwarder addresses
+    # that issue by transforming String keys into Symbols for the sake of matching
+    # against them:
+    #
+    # ```ruby
+    # Dio.string_hash({ 'a' => 1 }) in { a: 1 }
+    # # => true
+    # ```
+    #
+    # @author [baweaver]
+    # @since 0.0.3
+    #
+    class StringHashForwarder < BaseForwarder
+      # If an Array is provided from nested values we may want to match against
+      # it as well. Wrap the values in new forwarders to accomodate this.
+      #
+      # @raises [Dio::Errors::NoDeconstructionMethod]
+      #   Debating on this one. If the base is not an Array it complicates how
+      #   to Array match against a Hash. Raise an error for now, consider
+      #   revisiting later.
+      #
+      # @return [Array]
+      def deconstruct
+        return @base_object.map(&NEW_DIVE) if @base_object.is_a?(Array)
```

Comment:
Super nit and probably doesn't matter, but wondering if `@base_object.instance_of?(Array)` might be better? Would you want to map even if it was a subclass of Array?